src/main/java/org/alfresco/rest/api/impl/ActionsImpl.java

```diff
@@ -144,6 +182,157 @@ public void setPrefixResolver(NamespacePrefixResolver prefixResolver)
                 actionDefinitions.size());
     }
 
+    @Override
+    public Action executeAction(Action action, Parameters parameters)
+    {
+
+        if (action == null)
+        {
+            throw new InvalidArgumentException("action is null");
+        }
+
+        // Check action definition.
+
+        if (action.getActionDefinitionId() == null || action.getActionDefinitionId().isEmpty())
+        {
+            throw new InvalidArgumentException("action.actionDefinitionId is null or empty");
+        }
+
+        org.alfresco.service.cmr.action.ActionDefinition actionDef = null;
+        try
+        {
+            actionDef = actionService.getActionDefinition(action.getActionDefinitionId());
+        }
+        catch (NoSuchBeanDefinitionException nsbdx)
+        {
+            // Intentionally empty.
+        }
+
+        if (actionDef == null)
+        {
+            throw new EntityNotFoundException(action.getActionDefinitionId());
```

Comment:
Why have the intentionally empty catch block above and then test for a null `actionDef` when you could move the `throw new EntityNotFoundException(action.getActionDefinitionId());` into the catch? It would be less code.

Reply:
The empty catch was intentionally added to catch the case when the bean is found but it isn't an instance of ActionExecuter. This is the only case when the result of getActionDefinition can be null and not throw the exception.

Reply:
OK, that makes sense, a comment might be good on the `actionDef == null` to stop someone erroneously "optimising" that logic out ;-)

Reply:
I will add the comment.